README.md

```diff
@@ -1,126 +1,198 @@
+| Variable  | Type | Required | Description |
+| ------------- | ------------- | --- |------------- |
+| 	variable          | string  | true    |  define the variable name specified in the `values.yaml`file, using `foo.bar` for nested object. |
+| 	label             | string  | true      |  define the UI label. |
+| 	description       | string  | false      |  specify the description of the variable.|
+| 	type              | string  | false      |  default to `string` if not specified (current supported types are string, multiline, boolean, int, enum, password, storageclass, hostname, pvc, and secret).|
+| 	required          | bool    | false      |  define if the variable is required or not (true \| false)|
+| 	default           | string  | false      |  specify the default value. |
+| 	group             | string  | false      |  group questions by input value. |
+| 	min_length        | int     | false      | min character length.|
+| 	max_length        | int     | false      | max character length.|
+| 	min               | int     | false      |  min integer length. |
+| 	max               | int     | false      |  max integer length. |
+| 	options           | []string | false     |  specify the options when the vriable type is `enum`, for example: options:<br> - "ClusterIP" <br> - "NodePort" <br> - "LoadBalancer"|
+| 	valid_chars       | string   | false     |  regular expression for input chars validation. |
+| 	invalid_chars     | string   | false     |  regular expression for invalid input chars validation.|
+| 	subquestions      | []subquestion | false|  add an array of subquestions.|
+| 	show_if           | string      | false  | show current variable if conditional variable is true, for example `show_if: "serviceType=Nodeport"` |
+| 	show\_subquestion_if |  string  | false     | show subquestions if is true or equal to one of the options. for example `show_subquestion_if: "true"`|
 
-3. You can directly add `https://github.com/rancher/partner-charts` into rancher catalog. In order to show all the charts you have to run `make CHART=${chart_name} prepare` and make sure there is `chart-original` folder on each chart folder if your chart relies on a upstream chart.
+**subquestions**: `subquestions[]` cannot contain `subquestions` or `show_subquestions_if` keys, but all other keys in the above table are supported. 
 
-### Makefile
+### 6. Test your changes
 
-`make bootstrap`: 
+#### Generate modified chart
 
-Download binaries that are needed for ci scripts.
+To test your modified chart you will first need to generate it. The result is a tarball containing your upstream chart 
+with patch changes applied, and overlay files added on top of it. It will be located in the `assets` directory and as mentioned above, its name will be in the form `{CHART_NAME}-{VERSION}{packageVersion}.tgz`. 
```

Comment:
```suggestion
Run `make validate CHART={CHART_NAME}` to validate the chart and generate a tarball of your modified chart. This will create two directories:
- `charts/{CHART_NAME}/`: contains an unarchived version of your modified chart
- `assets/{CHART_NAME}/: contains an archived (tarball) version of your modified chart named `{CHART_NAME}-{VERSION}{packageVersion}.tgz`

It will also create several files (e.g. `index.html`, `index.yaml`, etc.) to set up a Helm repo in your current branch.

Alternatively, to skip validation, you can just run `make charts CHART={CHART_NAME}`.

To test your changes, just push the generated files to your fork as a separate commit and add your fork / branch as a Repository in the Dashboard UI. Your chart will then show up as an App in `Apps & Marketplace` under the Repository that you configured.

Make sure that you revert the generated files commit before submitting a PR!
```